owncloudApp/src/main/java/com/owncloud/android/operations/SynchronizeFolderOperation.java

```diff
@@ -272,30 +292,30 @@ private void removeLocalFolder() {
     }
 
     /**
-     *  Synchronizes the data retrieved from the server about the contents of the target folder
-     *  with the current data in the local database.
+     * Synchronizes the data retrieved from the server about the contents of the target folder
+     * with the current data in the local database.
      *
-     *  Grants that mFoldersToVisit is updated with fresh data after execution.
+     * Grants that mFoldersToVisit is updated with fresh data after execution.
      *
-     *  @param folderAndFiles   Remote folder and children files in folder
+     * @param remoteFolderAndFiles Remote folder and children files in folder
      */
-    private void mergeRemoteFolder(ArrayList<RemoteFile> folderAndFiles)
+    private void mergeRemoteFolder(ArrayList<RemoteFile> remoteFolderAndFiles)
             throws OperationCancelledException {
         Log_OC.d(TAG, "Synchronizing " + mAccount.name + mRemotePath);
 
         FileDataStorageManager storageManager = getStorageManager();
 
         // parse data from remote folder
-        OCFile updatedFolder = FileStorageUtils.createOCFileFrom(
-                (RemoteFile) folderAndFiles.get(0)
+        OCFile updatedFolder = FileStorageUtils.createOCFileFromRemoteFile(
+                remoteFolderAndFiles.get(0)
```

Comment:
Are you sure, index 0 does exists? Is there a check needed, if this element is available?

Reply:
`remoteFolderAndFiles` data comes from [`fetchFolderResult.getData()`](https://github.com/owncloud/android/pull/2559/files#diff-81259bfca184e3d1e468d72ab0e64ef0R226) call, whose result should contain always the root folder when [`fetchFolderResult.isSuccess()`](https://github.com/owncloud/android/pull/2559/files#diff-81259bfca184e3d1e468d72ab0e64ef0R224)